src/js/view/controls/components/timeslider.js

```diff
@@ -280,12 +280,14 @@ class TimeSlider extends Slider {
     }
 
     addCues(model, cues) {
-        this.resetChapters();
+        // Only reset chapters if setCues() is called with empty array (to clear cues)
```

Comment:
1. Can we rename `addCues` to set or update cues?
2. Shouldn't you always reset (`resetChapters` is really `resetCues` no?) before adding each new cue?